Research Examines!

GunHog opened the pull request:
Science Goggles and the Prototype Hardsuit Helmet may now scan items for
their research potential and machines for their contents.

Forum thread: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=4823

Examine an item to gain additional data.
- Tech levels
- Materials that can be salvaged from it if recycled
- Reliability rating
- Crit failure, if present.

Examine a machine to learn the components used to build it.

<b>Additional items with the ability may be added. Certain silicons/drones/swarmers may also be considered. 
EDIT: Borgs and Drones added, also now an action button!
EDIT 2: Tech names fixed!</b>

![screenshot 2015-10-07 13 39 18](https://cloud.githubusercontent.com/assets/4955510/10347657/9ed05128-6cfa-11e5-8ef7-40b70d730cbe.png)
<b> Machine Scanning </b>
![screenshot 2015-10-07 16 11 07](https://cloud.githubusercontent.com/assets/4955510/10351350/b2f6ad0a-6d0e-11e5-9cbc-4d5ffc7ef192.png)

<b>Known issues:</b>
- ~~Removing the RD hardsuit helm or Science Goggles while wearing the other causes the scanner var to be removed when it should not have been.~~
  ~~Tech names are not consistent with those given in the Protolathe.~~

Commit:
Research Examines!

Science Goggles and the Prototype Hardsuit Helmet may now scan items for
their research potential and machines for their contents.

Examine an item to gain additional data.
- Tech levels
- Materials that can be salvaged from it if recycled
- Reliability rating
- Crit failure, if present.

Examine a machine to learn the components used to build it.

RemieRichards commented on code/game/machinery/machinery.dm:
```diff
@@ -406,14 +404,22 @@ Class Procs:
+/obj/machinery/proc/display_parts(mob/user)
+	user << "<span class='notice'>Following parts detected in the machine:</span>"
+	for(var/var/obj/item/C in component_parts)
+		user << "<span class='notice'>   [C.name]</span>"
```
```\icon[C] [C.name]``` so they get a nice little picture too.

AlexCrimson commented:
Id say remove the tech numbers and just have the research type. Maybe give the more detailed examine to RD suit only.

GunHog commented:
I will have to make Science Goggles printable eventually. But, why remove the levels? That is the information someone wants most when looking for things to DA. As for the RD's hardsuit having the full examine only, I cannot justify that unless I manage to get it added to the primary map.

Commit:
Icons on component scan

- Because RR likes pretty pictures.
- Also closes a span properly.

Durkel commented:
Cool idea, Though I think we should just remove durability of items. Annoying having to decon eight+ bluespace cells so the damn thing don't break in a mech.

GunHog commented:
I am not too beg a fan of it either, but so long as it exists, I will have support for it. Only the AE gun and cells really use it.

WJohn commented:
More player feedback is always nice :+1:

Bawhoppen commented:
Don't give it to the prototype suit, I don't think that fits IMO, you should need to wear specific goggles if you want to see.

Xhuis commented:
Seems pretty good. You might want to have an on/off button for this, though, since I could see it getting annoying as chemist getting the glass count of a beaker whenever you examine it to see its reagents.

Commit:
Action buttons + More mobs!

- Scanner control is now an action button.
- Fixed multiple scanner bugs (I think)
- Gave ability to drones and borgs.

GunHog commented:
I have taken your feedback into account, and made it into an action button! It has the plus of having fixed the multiple scanners bug.

I have also given it to Cyborgs and Drones.

![screenshot 2015-10-08 11 32 08](https://cloud.githubusercontent.com/assets/4955510/10372981/52661e10-6db0-11e5-8a56-04dac9a5ab22.png)

Commit:
Buildable Science goggles

- You can now build Science Goggles at a protolathe.
- Fixes small var derp

Commit:
Proper name extraction

Uses the Protolathe procs to obtain the correct tech and material names!

Commit:
Offering to the Dark Lord

- Changelog

GunHog commented:
Changelog added!

Xhuis commented:
I'm not sure @GunHog wanted this merged...

GunHog commented:
I did, that is why I removed the WIP tag :)  I did add your feedback, Xhuis. The only change I would have had left to make is to get a better sprite, or add more functions/objects. Nothing required to make the feature work.

If you have additional questions or concerns, let me know and I can make any needed changes in a new PR.
